test/jacobian.jl

```diff
@@ -26,4 +26,52 @@ end
     else
         @test J(current_state(ds), current_parameters(ds), 0.0) == result
     end
-end
\ No newline at end of file
+end
+
+@testset "MTK Jacobian" begin
+    using ModelingToolkit
+    using ModelingToolkit: Num, RuntimeGeneratedFunctions.RuntimeGeneratedFunction
+    using DynamicalSystemsBase: SciMLBase
+    @independent_variables t
+    @variables u(t)[1:2]
+    D = Differential(t)
+    p = 3.0
+
+    eqs = [D(u[1]) ~ p * u[1],
+        D(u[2]) ~ -p * u[2]]
+
+    @named sys = ODESystem(eqs, t)
+    sys = structural_simplify(sys)
+
+    jac = calculate_jacobian(sys)
+    @test jac isa Matrix{Num}
+
+    prob = ODEProblem(sys, [1.0, 1.0], (0.0, 1.0); jac=true)
+    ode = CoupledODEs(prob)
+    @test ode.integ.f.jac.jac_oop isa RuntimeGeneratedFunction
+    @test ode.integ.f.jac([1.0, 1.0], [3.0], 0.0) isa Matrix{Float64}
```

Comment:
Tests must ONLY use the public API. Not access internal fields that may or may not exist. You have to re-write the tests to use the `jacobian` function exclusively after you create the `ode` type.